measures/ResStockArguments/measure.rb

```diff
@@ -785,6 +865,45 @@ def modify_setpoint_schedule(schedule, offset_magnitude, offset_schedule)
     return schedule
   end
 
+  def get_heating_and_cooling_seasons(args, epw_file, weather)
+    latitude = args[:site_latitude]
+    latitude = nil if latitude == Constants.Auto
+    latitude = HPXMLDefaults.get_default_latitude(latitude, epw_file)
+
+    heating_months, cooling_months = HVAC.get_default_heating_and_cooling_seasons(weather, latitude)
+    sim_calendar_year = Location.get_sim_calendar_year(nil, epw_file)
+
+    return heating_months, cooling_months, sim_calendar_year
+  end
+
+  def get_subset_begin_end_day_num(building_id, n_days, begin_day_num, end_day_num, year)
+    if begin_day_num > end_day_num
+      num_days = Constants.NumDaysInYear(year)
+      day_nums = (begin_day_num..num_days).to_a + (1..end_day_num).to_a
+    else
+      day_nums = (begin_day_num..end_day_num).to_a
+    end
+
+    begin_day_nums = day_nums[0...(-n_days - 1)]
+    if begin_day_nums.empty? # if n_days is longer than the season, set equal to the whole season
```

Comment:
Not too sure about this here. This is empty if the number of sampled unavailable days exceeds the number of days in the BA season. In this event, what should we do? It's currently coded up to just equal the BA season.